### PR TITLE
runtimes/d8.js: introduce V8's own harness agent (includes V8 license)

### PR DESCRIPTION
**Pull request by rwaldron**
This is blocked by https://github.com/bterlson/eshost/pull/50

### CODE REVIEW

File: runtimes/d8.js [hunk 1]
```diff
@@ -38,4 +38,109 @@ var $ = {
...
+              (0, eval)(\`${script}\`);
+              break;
+            case 'broadcast':
+              Atomics.add(i32a, ${BROADCAST_LOC}, 1);
+              broadcasts.push([msg.sab, msg.id]);
+              handleBroadcast();
+              break;
+          }
+        };
+        var $262 = {
```
Comment (bterlson) [inline runtimes/d8.js]: Looking at this skeptically...

Comment (rwaldron) [reply runtimes/d8.js]: Here's an example of how this is being used in practice: https://github.com/tc39/test262/blob/ce9419779fc61d5cdfa37a5cda7b6828c67221da/test/built-ins/Atomics/wait/did-timeout.js#L13-L20

File: runtimes/d8.js [hunk 2]
```diff
@@ -38,4 +38,109 @@ var $ = {
...
+            report(msg) {
+              postMessage(String(msg));
+              Atomics.add(i32a, ${WORKER_REPORT_LOC} + index, 1);
+            },
+            sleep(s) { Atomics.wait(i32a, ${SLEEP_LOC}, 0, s); },
+            leaving() {}
+          }
+        };`;
+    }
+
```
Comment (nomadtechie) [inline runtimes/d8.js]: @rwaldron - what limitations (if any) does the use of `Atomics` within the agent code here have on being able to cleanly use the v8 agent when running tests for `Atomics.wait` and `Atomics.add` in `test262-harness` ?

Comment (nomadtechie) [reply runtimes/d8.js]: Discuss this in person with @rwaldron and @jugglinmike - the conclusion was that while this isn't an ideal implementation - we have a uniform API for interacting with agents in test262 and this is an implementation detail which does not affect that API.

**Pull request merged**